src/language-usability.rst

```diff
@@ -16,6 +16,193 @@ Usability of Type Systems
+* Consistency - the design of the API should be consistent.
+* Matching Mental Models - the API user should have an accurate mental model of how the API executes procedures.
+
+.. container:: bib-item
+
+  .. bibliography:: api-usability.bib
+    :filter: key == 'myers2016improving'
+
+  This paper provides an overview of the design aspects of API design
+  (Learnability, Productivity, Error-Prevention, Simplicity, Consistency, and Matching Mental Models)
+  and the stakeholders (API designer, API user, and customer) involved with each design aspect.
+  This paper provides a high level survey on human-centered methods for designing APIs,
+  evaluating APIs, mitigating problems in APIs, and tools for improving API usability.
+
+.. container:: bib-item
+
+  .. bibliography:: api-usability.bib
+    :filter: key == 'scheller2015automated'
+
+  This paper identifies key aspects in API usability, in an *API Concepts Framework*,
+  a structure where *concepts* represent possible user actions, leading
+  to measureable properties on usability. Using this framework, the paper introduces
+  an automated method for measuring API usability, which they test in a user study and
+  find their calculated values highly correlated with user measured time.
+
+User Studies
+------------
+
+..
+  Outline
+  * User study studying global usability of APIs very difficult, so many factors
+  * Most user studies in API examine a specific aspect of api design and test it
+  * Paper: Usability Implications of Requiring Parameters in Objects' Constructors
+  * Paper: The Implications of Method Placement on API Learnability
+..
+
+Testing API design is difficult, as the numerous API design
+decisions all affect the overall usability interdependently
+:cite:`myers2016improving`. Testing the high-level and global design decisions
+is often intractable because of this. Most user studies in API usability
+examine a specific aspect of API design :cite:`stylos2007usability`
+:cite:`stylos2008implications` :cite:`ellis2007factory`.
+
+.. container:: bib-item
+
+  .. bibliography:: api-usability.bib
+    :filter: key == 'stylos2007usability'
+
+  This paper examines the usability implications of class constructor parameters.
+  The study in this paper showed that developers
+  strongly preferred and were more effective with APIs that did not
+  require parameters in object constructors. This contradicted the
+  belief that object constructor parameters would "self-document"
+  themselves and guide developers to using the objects correctly.
+
+.. container:: bib-item
+
+  .. bibliography:: api-usability.bib
+    :filter: key == 'stylos2008implications'
+
+  This paper examines the impact of method placement on usability.
+  In the user study participants were given nearly identical APIs but
+  the comparison group contained a method that was placed in an
+  unexpected location. Participants with "correctly-placed" methods
+  were orders of magnitudes faster in their tasks.
```

Comment:
"orders of magnitudes" is vague -- can you provide some numbers? what are "their tasks"?